storage/ndb/src/kernel/blocks/dbtup/AggInterpreter.cpp

```diff
@@ -0,0 +1,1572 @@
+                       n_agg_results_ * sizeof(AggResItem);
+#ifdef MOZ_AGG_MALLOC
+      agg_rec = MemAlloc(len_in_char +
+                          n_agg_results_ * sizeof(AggResItem));
+#else
+      agg_rec = new char[len_in_char +
+                        n_agg_results_ * sizeof(AggResItem)];
+#endif // MOZ_AGG_MALLOC
+      memset(agg_rec, 0, len_in_char +
+                        n_agg_results_ * sizeof(AggResItem));
+      memcpy(agg_rec, reinterpret_cast<char*>(buf_), len_in_char);
+      GBHashEntry new_entry{agg_rec, len_in_char};
+
+      gb_map_->insert(std::make_pair<GBHashEntry, GBHashEntry>(
+                      std::move(new_entry), std::move(
+            GBHashEntry{agg_rec + len_in_char,
+            static_cast<uint32_t>(n_agg_results_ * sizeof(AggResItem))})));
+      n_groups_ = gb_map_->size();
+      agg_res_ptr = reinterpret_cast<AggResItem*>(agg_rec + len_in_char);
+
+      for (uint32_t i = 0; i < n_agg_results_; i++) {
+        agg_res_ptr[i].type = agg_results_[i].type;
+      }
+    }
+  } else {
+    agg_res_ptr = agg_results_;
+  }
+
+  uint32_t col_index;
+
+  uint32_t value;
+  DataType type;
+  bool is_unsigned;
+  uint32_t reg_index;
+
+  uint32_t reg_index2;
+
+  uint32_t agg_index;
+
+  const uint32_t* attrDescriptor = nullptr;
+
+  int32_t decimal_info = 0;
+  int32_t precision = 0;
+  int32_t scale = 0;
+  decimal_t decimal;
+  decimal.buf = decimal_buf_;
+  int32_t dec_ret = E_DEC_OK;
+  uint8_t* dec_buf_ptr = nullptr;
+  longlong dec_val_ll = 0;
+  ulonglong dec_val_ull = 0;
+
+  uint32_t exec_pos = agg_prog_start_pos_;
+  while (exec_pos < prog_len_) {
+    value = prog_[exec_pos++];
+    uint8_t op = (value & 0xFC000000) >> 26;
+    int ret = 0;
+    buf_pos_ = 0;
+    AttributeHeader* header = nullptr;
+
+    switch (op) {
+      case kOpPlus:
+        reg_index = (value >> 12) & 0x0F;
+        reg_index2 = (value >> 8) & 0x0F;
+
+        ret = RegPlusReg(registers_[reg_index], registers_[reg_index2],
+                  &registers_[reg_index]);
+        // assert(ret >= 0);
+        if (ret < 0) {
+          g_eventLogger->warning("Overflow[PLUS], value is out of range");
+          return ZAGG_MATH_OVERFLOW;
+        }
+        break;
+      case kOpMinus:
+        reg_index = (value >> 12) & 0x0F;
+        reg_index2 = (value >> 8) & 0x0F;
+
+        ret = RegMinusReg(registers_[reg_index], registers_[reg_index2],
+                  &registers_[reg_index]);
+        // assert(ret >= 0);
+        if (ret < 0) {
+          g_eventLogger->warning("Overflow[MINUS], value is out of range");
+          return ZAGG_MATH_OVERFLOW;
+        }
+        break;
+      case kOpMul:
+        reg_index = (value >> 12) & 0x0F;
+        reg_index2 = (value >> 8) & 0x0F;
+
+        ret = RegMulReg(registers_[reg_index], registers_[reg_index2],
+                  &registers_[reg_index]);
+        // assert(ret >= 0);
+        if (ret < 0) {
+          g_eventLogger->warning("Overflow[MUL], value is out of range");
```

Comment:
Change to g_eventLogger->debug, pretty sure warnings gets printed by default whereas debug only gets printed when specifically activated

Reply:
Good compromise between Alibaba style and RonDB style